### ISSUE TITLE
XCB event handling broken with multiple windows

### ISSUE THREAD

**Issue by binary1248** [LaurentGomila/SFML#843]
If an application has multiple windows, event handling will break since the events are handled per-connection (which is shared) and not per-window. Any window can handle any event destined for any window. This leads to broken behaviour such as clicking the close button of one window closing another window instead.

This might have already been the case with the previous Xlib implementation, but I didn't bother to test that.

The fix would be to make windows inspect events for the id of the window they are destined for and queue events that don't belong to them to an event buffer that each window would have to have.

Originally reported on forum: http://en.sfml-dev.org/forums/index.php?topic=17780.0

Test code:
```cpp
#include <SFML/Window.hpp>

int main()
{
    sf::Event event;
    sf::Window* windows[9];

    for (int i = 0; i < 9; ++i)
        windows[i] = new sf::Window(sf::VideoMode(80, 60), "");

    int windowCount = 9;

    while (windowCount > 0)
        for (int i = 0; i < 9; ++i)
            if (windows[i]->isOpen())
                while (windows[i]->pollEvent(event))
                    if (event.type == sf::Event::Closed)
                    {
                        windows[i]->close();
                        --windowCount;
                    }

    for (int i = 0; i < 9; ++i)
        delete windows[i];
}
```

**Comment by LaurentGomila** [comment]
> This might have already been the case with the previous Xlib implementation, but I didn't bother to test that.

Nop, the X11 implementation correctly filtered events by window id.
https://github.com/LaurentGomila/SFML/blob/45810a134505f9d21d58f35f30d05d44c4a3db59/src/SFML/Window/Unix/WindowImplX11.cpp#L325

**Comment by binary1248** [comment]
Should be fixed in 386d4925b25f632294feaa63f5d1f83cae73c40c.

**Comment by gsfare** [comment]
Just to confirm, I checked out and built the branch, it is working fine for me now.

**Comment by binary1248** [comment]
Merged in b9cc6f5c0de4587c70ae9447a1d7a501cefaded1
